### ISSUE TITLE
postinst craps over existing server_name setting

### ISSUE THREAD

**Issue by richvdh** [matrix-org/package-synapse-debian#21]
If `server_name.yaml` is edited manually, the postinst overwrites it on upgrade. A few points:
* if it's autogenerated, the file should say so
* an upgrade shouldn't touch it anyway (the postinst can tell if it's an upgrade or a fresh install)
* even on fresh install, if the file exists, maybe we should leave it alone?

**Comment by gerco** [comment]
I agree, it shouldn't clobber the file if you manually edited it. As a workaround though, you can update the debconf database with the correct name so future updates won't break it again:

To fix your servername in the debconf database, use this:
`root@matrix:~# echo "set matrix-synapse/server-name your.server.name" | debconf-communicate`

Then check if the new name took:
```
root@matrix:~# debconf-show matrix-synapse
* matrix-synapse/report-stats: true
* matrix-synapse/server-name: your.server.name
```

Next time, the postinst will put the correct name in the servername.yaml file.

**Comment by richvdh** [comment]
or `dpkg-reconfigure matrix-synapse`

**Comment by uhoreg** [comment]
> an upgrade shouldn't touch it anyway (the postinst can tell if it's an upgrade or a fresh install)

FWIW, this is also what the first sentence of [Debian Policy section 10.7.3](https://www.debian.org/doc/debian-policy/#behavior) says.

**Comment by richvdh** [comment]
Transferring this to synapse repo, since the packaging is maintained there (and it is still a problem in matrix-synapse-py3)
